### ISSUE TITLE
existing `/etc/sub*id` incorrectly updated by `nixos-rebuild switch` when `users.users.*.subUidRanges` and `users.users.*.subGidRanges` are explicitly set

### ISSUE THREAD

**Issue by ppenguin** [NixOS/nixpkgs#155581]
### Describe the bug

`/etc/subuid` and `/etc/subgid` are incorrectly updated by `nixos-rebuild switch` when `users.users.*.subUidRanges` and `users.users.*.subGidRanges` are explicitly set, which leads to users having multiple entries with different values in these files after a rebuild, ultimately leading to failing `podman` execution.

### Steps To Reproduce
1. observe pre-existing `/etc/sub*id` files with IDs defined for the "normal users" defined in `users.users.*`
2. add definitions `users.users.*.sub*idRanges` for said users to enforce explicitly defined IDs also for new users
3. `sudo nixos-rebuild switch`
4. observe double entries for each defined user in `/etc/sub*id` with different IDs and failing userspace execution of `podman run`

### Expected behavior

Declarative definition of `users.users.*.sub*idRanges` **replaces** corresponding values in `/etc/sub*id` instead of **adding** them

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.10, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.5.0pre20211206_d1aaa7e`
 - channels(root): `"home-manager-21.11, musnix, nixos-21.11.335331.8a70a6808c8"`
 - channels(mirko): `"home-manager-21.11"`
 - channels(bjoern): `"home-manager-21.11"`
 - channels(uemit): `"home-manager-21.11"`
 - channels(jeroen): `"home-manager-21.11, unstable-22.05pre336340.cdaa4ce25b7"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by thomasjm** [comment]
I wanted to make sure this wasn't happening on my system but I couldn't reproduce it.

I tried the following:
* Add `users.users.*.sub*idRanges` for a couple existing users and verify the `/etc/sub*id` files are created correctly.
* Change the range for one of those users and verify the files were updated correctly.
* Add `sub*idRanges` to an additional user and verify the files were updated correctly.
* Remove the ranges from the additional user and verify the files were updated correctly.

Maybe I'm not understanding where the "pre-existing" `/etc/sub*id` files came from in this report. But maybe it's been fixed?

**Comment by ppenguin** [comment]
I just tried again after just deleting `/etc/sub*id` but a bit different (more convenient) use case using `autoSubUidGidRange`:
- I just chose a few `podman` users and set `autoSubUidGidRange = true;`
- for some other users I set `autoSubUidGidRange = false;`

This creates `subids` only for the first group of users as expected, BUT one of them is a system user (`gitlab-runner`) and sure enough has overlapping IDs with a normal user.

An additional system user `isSystemUser = true` with auto UidGid creates an additional overlapping ID.

This is different, I couldn't reproduce the earlier case on current nixos.

So I'll close this for now and file a new bug for the auto ID issue.

https://github.com/NixOS/nixpkgs/issues/259588
